cranelift/filetests/filetests/isa/x64/struct-ret.clif

```diff
@@ -95,7 +96,8 @@ block0(v0: i64):
 ;   subq $0x10, %rsp
 ;   movq %r15, (%rsp)
 ; block1: ; offset 0xc
-;   movq %rdi, %r15
+;   movq %rdi, %rax
+;   movq %rax, %r15
```

Comment:
This is a little surprising, but also a totally reasonable consequence of no longer communicating program moves to RA2.

Reply:
This was easily resolved by copying the sret register in `Lower::new`, instead of generating a temporary that we would move into later.